### PR TITLE
DHFPROD-8710: Save Step Settings 5.7

### PR DESCRIPTION
**Pull request by MSBrignoni**
### Description
Save Step Settings

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests
- [x] Added to Release Wiki

### CODE REVIEW

**Review by briantang — DISMISSED** [review]
Looks good to me!

File: marklogic-data-hub-central/ui/src/components/flows/flows.tsx [hunk 1]
```diff
@@ -554,6 +572,80 @@ const Flows: React.FC<Props> = ({
     }
   };
 
+  const getLocalStorageDataUser = () => {
+
+    if (getUserPreferencesLS()) {
```
Comment (xnikhil08) [inline marklogic-data-hub-central/ui/src/components/flows/flows.tsx:577]: Same suggestion here. Let's get the return value of the function getUserPreferencesLS() before line 577 here and then use it further down inside the if logic.

File: marklogic-data-hub-central/ui/src/components/flows/flows.tsx [hunk 2]
```diff
@@ -554,6 +572,80 @@ const Flows: React.FC<Props> = ({
...
+      } else { return false; }
+    }
+
+    return oldOptions;
+  };
+
+  const saveLocalStoragePreferences = (value: boolean, saveSelectedStepsDataUser?: boolean) => {
+
+    const sessionUser = localStorage.getItem("dataHubUser");
+    if (getUserPreferencesLS()) {
```
Comment (xnikhil08) [inline marklogic-data-hub-central/ui/src/components/flows/flows.tsx:617]: Again the same suggestion. Let's avoid fetching the userPreferences multiple times. Try fetching them once at the beginning of the current scope and then use the return value further as needed, within the same scope.

File: marklogic-data-hub-central/ui/src/components/flows/flows.tsx [hunk 3]
```diff
@@ -588,6 +680,8 @@ const Flows: React.FC<Props> = ({
     } else {
       onCheckboxChange("default", step.stepName, step.stepNumber, step.stepDefinitionType, flowNameCheck, step.stepId, step.sourceFormat);
     }
+    // if (getUserPreferencesLS() && getUserPreferencesLS()?.loadSelectedStepsUser) getLocalStorageDataUser();
```
Comment (xnikhil08) [inline marklogic-data-hub-central/ui/src/components/flows/flows.tsx:683]: Let's remove the comments if the code is not needed anymore.

**Review by xnikhil08 — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/ui/src/components/flows/flows.tsx (hunk 3 above)
Comment (MSBrignoni) [reply marklogic-data-hub-central/ui/src/components/flows/flows.tsx]: Thanks, done.

**Review by xnikhil08 — APPROVED** [review]

**Review by briantang — APPROVED** [review]

**Pull request merged**